activesupport/lib/active_support/callbacks.rb

```diff
@@ -747,12 +803,25 @@ def define_callbacks(*names)
           options = names.extract_options!
 
           names.each do |name|
-            class_attribute "_#{name}_callbacks", instance_writer: false
+            name = name.to_sym
+
             set_callbacks name, CallbackChain.new(name, options)
 
             module_eval <<-RUBY, __FILE__, __LINE__ + 1
```

Comment:
There is [one mention](https://github.com/rails/rails/blob/e1aabeedd896fe381570f85c5b4116049c49f1b6/activerecord/lib/active_record/callbacks.rb#L240), but maybe it's up to AR to provide and deprecate those.